api/src/main/java/io/opentelemetry/OpenTelemetry.java

```diff
@@ -183,6 +198,40 @@ private OpenTelemetry() {
     return null;
   }
 
+  /**
+   * Sets up the OpenTelemetry API to use the given implementation objects.
+   *
+   * <p>This method must not be called after attempting to access an API implementation object via
+   * this class (i.e., after calling methods such as {@link #getTracerProvider()}). Basically this
+   * means that the one who is in control of the main method is the only one who may call this
+   * method. Consequently, If another method calls this, this requirement applies to that method
+   * too, recursively.
+   *
+   * <p>This method must only be called once, even if you use any {@code null} arguments. Any null
+   * arguments are defaulted using the usual mechanism, falling back to the default (no-op)
+   * implementations if no SPI can be found.
+   *
+   * @param tracerProvider The {@link TracerProvider} instance to use, or {@code null} to load using
+   *     default mechanism.
+   * @param meterProvider The {@link MeterProvider} instance to use, or {@code null} to load using
+   *     default mechanism.
+   * @param contextManager The {@link CorrelationContextManager} instance to use, or {@code null} to
+   *     load using default mechanism.
+   * @throws IllegalStateException if OpenTelemetry is already initialized.
+   */
+  public static void setupExplicitImplementations(
+      @Nullable TracerProvider tracerProvider,
+      @Nullable MeterProvider meterProvider,
+      @Nullable CorrelationContextManager contextManager) {
+    synchronized (OpenTelemetry.class) { // No double-checked locking, not performance-sensitive.
+      if (instance == null) {
+        instance = new OpenTelemetry(tracerProvider, meterProvider, contextManager);
+        return;
+      }
+    }
+    throw new IllegalStateException("OpenTelemetry is already initialized.");
```

Comment:
Since this is a part of the API, should we log this, rather than throwing an exception that could bring down the process?

Reply:
Throwing is OK in the initialization phase.

Reply:
The spec says that throwing **on initialization** is alright:
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/error-handling.md#basic-error-handling-principles

I'd also be fine with defaulting to SPI and just logging as well, however.

Reply:
This might not happen during initialization. In fact, it only throws *after* initialization is complete!  :)  I'd be more comfortable with logging, but I don't feel super strongly about it.

Reply:
getInstance may also throw, and this one can be invoked by getTracer and this one is even usually called after initialization. Which, actually, is something that should be improved 🤔  But rather not in this PR.